src/app/projects/states/dashboard/directives/task-dashboard/directives/task-status-card/task-status-card.coffee

```diff
@@ -43,10 +43,24 @@ angular.module('doubtfire.projects.states.dashboard.directives.task-dashboard.di
           "This task is past the target due date, so you will need to request an extension to get feedback. Use confirm to request extensions and proceed with the submission.",
           () -> extendAndSubmit(trigger))
       else
-        $scope.task.triggerTransition(trigger, $scope.unitRole)
+        $scope.triggerStatusChange(trigger)
 
     # Allow upload of new files for an updated submission
     $scope.updateFilesInSubmission = ->
       taskService.presentTaskSubmissionModal($scope.task, $scope.task.status, true)
 
+    $scope.triggerStatusChange = (trigger) ->
+      if (trigger == 'not_started' || trigger == 'working_on_it' || trigger == 'need_help') && $scope.task.status == 'ready_to_mark' && !$scope.task.isOverdue()
```

Comment:
We should have helper methods to determine this case, we don't want to have the state checks in the component like this, check the taskService and the `isInxxxState` methods.